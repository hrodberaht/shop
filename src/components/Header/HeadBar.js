import React, { Component } from 'react';

export default class HeadBar extends Component {
  render() {
    return (
      <div className="headbar">
        <hr />
      </div>
    );
  }
}
