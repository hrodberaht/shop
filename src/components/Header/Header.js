import React, { Component } from 'react';
import ConnectedSideBar from './SideBar';

export default class Header extends Component {
  render() {
    return <ConnectedSideBar />;
  }
}
