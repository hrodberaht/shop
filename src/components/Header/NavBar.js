import React, { Component } from 'react';
import { NavLink, Link } from 'react-router-dom';
import ConnectedAuthorization from '../Auth/Authorization';

export default class NavBar extends Component {
  render() {
    return (
      <div className="nav-bar">
        <ul className="nav-bar__list">
          <li className="item">
            <NavLink to="/">Shop</NavLink>
          </li>
          <li className="item">
            <NavLink to="/orders">Orders</NavLink>
          </li>
          <li className="item">
            <NavLink to="/cart">Cart</NavLink>
          </li>
          <ConnectedAuthorization
            render
            withRoleAdmin={(
              <li className="item">
                <Link to="/admin">Admin</Link>
              </li>
)}
          />
          <ConnectedAuthorization
            logout={logout => (
              <li className="item">
                <Link to="/" onClick={logout}>
                  Logout
                </Link>
              </li>
            )}
          />
        </ul>
      </div>
    );
  }
}
