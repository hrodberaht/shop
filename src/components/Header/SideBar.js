import React, { Component } from 'react';
import { NavLink, Link, withRouter } from 'react-router-dom';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import { connect } from 'react-redux';
import ConnectedAuthorization from '../Auth/Authorization';
import { getNavigationSidebarValue } from '../../store/navigation/selectors';

export class SideBar extends Component {
  render() {
    const { toggleSidebar } = this.props;
    return (
      <div className={classNames({ sidebarHide: !toggleSidebar, sidebar: toggleSidebar })}>
        <ul className="sidebar__list">
          <li className="item">
            <Link to="/">Logo</Link>
          </li>
          <hr />
          <li className="item">
            <NavLink exact to="/">
              Shop
            </NavLink>
          </li>
          <li className="item">
            <NavLink to="/orders">Orders</NavLink>
          </li>
          <li className="item">
            <NavLink to="/cart">Cart</NavLink>
          </li>
          <li className="item">
            <NavLink to="/wishlist">Wish list</NavLink>
          </li>
          <ConnectedAuthorization
            render
            withRoleAdmin={(
              <li className="item">
                <NavLink to="/inventory">Inventory</NavLink>
              </li>
)}
          />
          <ConnectedAuthorization
            logout={logout => (
              <li className="item">
                <Link to="/" onClick={logout}>
                  Logout
                </Link>
              </li>
            )}
          />
        </ul>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  toggleSidebar: getNavigationSidebarValue(state),
});

export default withRouter(connect(mapStateToProps)(SideBar));

SideBar.propTypes = {
  toggleSidebar: PropTypes.bool.isRequired,
};
