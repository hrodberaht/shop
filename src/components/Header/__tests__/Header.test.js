import Header from '../Header';

const setup = buildSetup(Header, {});

describe('<Header />', () => {
  it('should render without crash', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot();
  });
});
