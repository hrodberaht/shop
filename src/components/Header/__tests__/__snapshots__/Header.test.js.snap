// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> should render without crash 1`] = `<Connect(SideBar) />`;
